### PR TITLE
fix: optimize scripts for parallel startup of Looker docker image

### PR DESCRIPTION
**Pull request by drstrangelooker**
I move the setup for getting and starting the Docker image to the start of the process. Then I let the other setup proceed.
Finally, just before the actual tests, I run the wait_for_looker script to check that Looker has started successfully.

### CODE REVIEW

File: .github/workflows/codegen-ci.yml [hunk 1]
```diff
@@ -32,13 +32,32 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+      - name: Repo Checkout
+        uses: actions/checkout@v2
+
```
Comment (jkaster) [inline .github/workflows/codegen-ci.yml:37]: does moving this up affect the responsiveness of canceling previous runs? Just wondering why it was moved

**Review by jkaster — APPROVED** [review]
LGTM. One question about moving one of the steps

File: .github/workflows/codegen-ci.yml (hunk 1 above)
Comment (drstrangelooker) [reply .github/workflows/codegen-ci.yml:37]: The repo checkout can overwrite some of the previous work that was done. I ran into a weird situation with the python-ci.yml where not having the Repo Checkout first caused the gcloud auth step to fail. So I decided to do it first for all of them just to make sure there wasn't a race condition.

Comment (jkaster) [reply .github/workflows/codegen-ci.yml:37]: Thanks. Sounds good

**Pull request merged**